File: momentra/momentra/Shell/PersonalEmpty/PersonalMasterExpenseTheme.swift
import SwiftUI

/// Figma 453:9376 Master Expense premium tokens — fixed purple accent.
enum PersonalMasterExpenseTheme {
    static let bg = Color(hex: "#191622")
    static let surface = Color(hex: "#181424").opacity(0.7)
    static let surfaceSolid = Color(hex: "#181424")
    static let border = Color(hex: "#342C44")
    static let accent = Color(hex: "#7C5CFC")
    static let accentLight = Color(hex: "#C9BFFF")
    static let textMain = Color(hex: "#E5E0EE")
    static let muted = Color(hex: "#9E9AA7")
    static let categoryUnselected = Color(hex: "#14131A")
    static let categoryBorder = Color.white.opacity(0.1)
    static let error = Color(hex: "#F87171")

    struct EmotionalOption: Identifiable {
        let id: String
        let emoji: String
        let label: String
    }

    struct RelationshipImpactOption: Identifiable {
        let id: String
        let emoji: String
        let label: String
    }

    static let emotionalOptions: [EmotionalOption] = [
        .init(id: "Relieved", emoji: "😌", label: "Relieved"),
        .init(id: "Happy", emoji: "😊", label: "Happy"),
        .init(id: "Connected", emoji: "🤝", label: "Connected"),
        .init(id: "Proud", emoji: "😎", label: "Proud"),
        .init(id: "Neutral", emoji: "😐", label: "Neutral"),
        .init(id: "Unsure", emoji: "🤔", label: "Unsure"),
        .init(id: "Guilty", emoji: "😔", label: "Guilty"),
        .init(id: "Blessed", emoji: "🙏", label: "Blessed"),
        .init(id: "Disappointed", emoji: "😞", label: "Disappointed"),
    ]

    static let sharedWithOptions = ["Spouse", "Parents", "Family", "Friends", "Custom"]

    static let relationshipImpactOptions: [RelationshipImpactOption] = [
        .init(id: "Strengthened Connection", emoji: "💪", label: "Strengthened Connection"),
        .init(id: "Celebration Together", emoji: "🎉", label: "Celebration Together"),
        .init(id: "Support Given", emoji: "🤲", label: "Support Given"),
    ]

    static let reasoningOptions = ["Celebration", "Daily Need", "Gift", "Travel", "Other"]
    static let segmentOptions = ["Low", "Medium", "High"]
    static let whenOptions = ["Now", "Today", "Yesterday"]
}

File: apk/app/src/main/java/com/example/momentra/ui/shell/personal/PersonalMasterExpenseTheme.kt
package com.example.momentra.ui.shell.personal

import androidx.compose.ui.graphics.Color

/** Figma 453:9376 Master Expense premium tokens — fixed purple accent. */
object PersonalMasterExpenseTheme {
    val Bg = Color(0xFF191622)
    val Surface = Color(0xB3181424)
    val SurfaceSolid = Color(0xFF181424)
    val Border = Color(0xFF342C44)
    val Accent = Color(0xFF7C5CFC)
    val AccentLight = Color(0xFFC9BFFF)
    val Text = Color(0xFFFFFFFF)
    val TextMain = Color(0xFFE5E0EE)
    val Muted = Color(0xFF9E9AA7)
    val CategoryUnselected = Color(0xFF14131A)
    val CategoryBorder = Color.White.copy(alpha = 0.1f)
    val Error = Color(0xFFF87171)

    data class EmotionalOption(val emoji: String, val label: String)
    data class RelationshipImpactOption(val emoji: String, val label: String)

    val emotionalOptions = listOf(
        EmotionalOption("😌", "Relieved"),
        EmotionalOption("😊", "Happy"),
        EmotionalOption("🤝", "Connected"),
        EmotionalOption("😎", "Proud"),
        EmotionalOption("😐", "Neutral"),
        EmotionalOption("🤔", "Unsure"),
        EmotionalOption("😔", "Guilty"),
        EmotionalOption("🙏", "Blessed"),
        EmotionalOption("😞", "Disappointed"),
    )

    val sharedWithOptions = listOf("Spouse", "Parents", "Family", "Friends", "Custom")

    val relationshipImpactOptions = listOf(
        RelationshipImpactOption("💪", "Strengthened Connection"),
        RelationshipImpactOption("🎉", "Celebration Together"),
        RelationshipImpactOption("🤲", "Support Given"),
    )

    val reasoningOptions = listOf("Celebration", "Daily Need", "Gift", "Travel", "Other")

    val segmentOptions = listOf("Low", "Medium", "High")
    val whenOptions = listOf("Now", "Today", "Yesterday")
}
